collections/CONTRIBUTING.md

```diff
@@ -0,0 +1,34 @@
+# Contribution Guide
+
+First, thank you for being interested in contributing! We hope this page helps as a guide. When in doubt, look at the files
+in this folder, read through issues and PRs regarding the `collections` module or visit the
+[#deno-std channel on our Discord](https://discord.gg/Ub9bRxHv) and ask.
+
+## Philosophy
+
+Currently, those are the general ideas behind how the module is implemented:
+
+- Provide a clear, specific toolbox and vocabulary for common tasks with `Array`s and `Record`s
+- Optimize each specific tasks for maximum runtime and memory efficiency
+- Offer only pure functions, never mutate given arguments, always return a new value
```

Comment:
I feel we are open to including mutating functions when it makes sense. For example, deepAssign suggestion seems worth the consideration in my view.

Reply:
Yes I agree. Maybe change to "functions are pure by default, side effect need to be clearly marked and putt into <folder>"?

Reply:
> Functions are expected to be pure by default, side effect (like mutation of arguments) need to be clearly documented and the function put into someFolderNameWeNeedToFigureOut

Sounds good to me.

Reply:
The folder name might be `collections/mutate`, `collections/mut`? I'm even ok with no special folder for mutating functions if it's documented well.

Reply:
I will change this today 👌